components/SearchResults/index.tsx

```diff
@@ -101,50 +73,71 @@ function SearchResults({ value }: SearchResultsProps) {
     lineHeight: 'calc(100vh - 72px - 53px - 68.5px)',
   };
 
-  const [isAddStock, setIsAddStock] = useState(false);
-  const handleAddStocks = () => {
-    {
-      isAddStock ? setIsAddStock(false) : setIsAddStock(true);
-    }
+  // Jotai의 selectedStocksAtom을 이용해서 선택된 주식을 관리
+  const [selectedStocks, setSelectedStocks] = useAtom(selectedStocksAtom);
+
+  console.log('getSearchStocks?.data.data: ', getSearchStocks?.data.data);
+  // 선택 상태를 토글하여 선택된 객체값 배열 핸들링
+  const handleToggleSelected = (stock: SelectedStocksAtomProps) => {
+    setSelectedStocks((prev: SelectedStocksAtomProps[]) => {
+      // 이미 선택된 주식인지 아닌지 확인하고, 선택 상태 토글
+      if (
+        prev.some(
+          (selected: SelectedStocksAtomProps) =>
+            selected.stockCode === stock.stockCode,
+        )
+      ) {
+        return prev.filter(
+          (selected: SelectedStocksAtomProps) =>
+            selected.stockCode !== stock.stockCode,
+        );
```

Comment:
여기도 마찬가지 어떤 값을 return 하는지 제가 코드를 다 봐야지만 알 수 있네요 😥
`prev라는 Array를 순회하면서 선택된것 중에서 stockCode끼리 비교를 해서 동일하지 않은 것들만 담은 새 배열을 return 한다.`

아래와 같이 변경하면 더 쉽게 return 값을 파악할 수 있을 것 같습니다.
```suggestion
const stockCodeNotMatches = prev.filter(
          (selected: SelectedStocksAtomProps) =>
            selected.stockCode !== stock.stockCode,
        );
        
return stockCodeNotMatches
```

Reply:
```typescript
      if (isNewSelectedStocks) {
        const notMatchedStocks = prev.filter(
          (selected: SelectedStocksAtomProps) =>
            selected.stockCode !== stock.stockCode,
        );

        return notMatchedStocks;
      } else {
        const newAddedSelectedStocks = [...prev, stock];

        return newAddedSelectedStocks;
      }
```
로 변경해봤습니다.